### ISSUE TITLE
Test failure in buffer-simplification

### ISSUE THREAD

**Issue by blishko** [ethereum/hevm#561]
One of our test is currently failing, the following should reproduce the test failure:
```
cabal run test -- --quickcheck-replay=728804 -p '/buffer-simplification/'
```

**Comment by blishko** [comment]
This is a duplicate of https://github.com/ethereum/hevm/issues/401

This test failure can be simplified to equivalence check of simplification of expression `CopySlice (Lit 0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff) (Lit 0x0) (Lit 0x2) (ConcreteBuf "a") (ConcreteBuf "")`.

As mentioned in #401, the problem is wraparound of `0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff` (2^256 - 1) in bit-vector semantics.
Our simplification says the result is a buffer containing two zero bytes.
But in SMT-LIB encoding, it reads the first byte from index 2^256 -1, which is 0, but the second byte is read from index 0 (because of wraparound of 2^256). which is **not** 0

**Comment by msooseth** [comment]
This is unfortunately very annoying and happens quite often. If you wanna fix it, I suggest taking a look at #401, and all the others who have already found it and thought about what could be done. Maybe it's an easy fix... would be SUPER nice if you could fix it!!

Closing, as duplicate of #401, but it's a good example and a good thought!
